src/util.test.js

```diff
@@ -0,0 +1,845 @@
+/*
+* util.js requires Oskari object with few functions in global variables in order to work without errors.
+* For this reason mock object is created and placed to global variables before importing util 
+*/
+const OskariMock = {
+    log: () => {
+        return {
+            // Prevent debug logging in tests to achieve leaner Jest test execution output
+            debug: (text) => { }
+        }
+    },
+    getDecimalSeparator: () => { return '.' }
+};
+
+global.Oskari = OskariMock;
+const util = require('./util');
+const MobileDetect = require('mobile-detect');
+const jQuery = require('jquery');
+global.MobileDetect = MobileDetect;
+global.jQuery = jQuery;
+
+describe('throttle function', () => {
+
+    test('executes function call only once within given wait parameter interval', () => {
+
+        expect.assertions(1);
+
+        const functionCallCount = 10;
+        const wait = 1000;
+        const mockFunction = jest.fn();
+
+        const throttledFunction = OskariMock.util.throttle(mockFunction, wait);
+
+        for (let i = 0; i < functionCallCount; i++) {
+            throttledFunction();
+        };
+        expect(mockFunction.mock.calls.length).toEqual(1);
+    });
+
+    test('executes all function calls when called with interval >= throttle wait parameter', async () => {
+
+        expect.assertions(1);
+
+        const functionCallCount = 5;
+        const wait = 200;
+        const mockFunction = jest.fn();
+
+        const throttledFunction = OskariMock.util.throttle(mockFunction, wait);
+
+        for (let i = 0; i < functionCallCount; i++) {
+            throttledFunction();
+            await sleep(wait);
```

Comment:
Shouldn't you first call the throttledFunction a number of times, check that it hasn't been called but once or 0 times, then sleep, then check that is has been called 5 times?

Reply:
Or add an assert on both sides of the sleep? This doesn't really test the throttling but that the function has been called 5 times after we call it 5 times?

Reply:
Mentioned test did not test actual throttling as you stated. I updated this test to indicate more clearly what is the purpose of that test. Updated tests should now test also actual throttling better .